src/Native/ObjWriter/objwriter.cpp

```diff
@@ -309,7 +310,13 @@ void ObjectWriter::SetCodeSectionAttribute(const char *SectionName,
 }
 
 void ObjectWriter::EmitAlignment(int ByteAlignment) {
-  Streamer->EmitValueToAlignment(ByteAlignment, 0x90 /* Nop */);
+  int64_t fillValue = 0x90; //x86 nop
+
+  if (TMachine->getTargetTriple().getArch() == llvm::Triple::ArchType::aarch64) {
```

Comment:
It may be more future proof to specifically check for x86/x64 here, and use 0 everywhere else.